manifests/server/service.pp

```diff
@@ -47,4 +47,18 @@
     File['mysql-config-file'] -> Service['mysqld']
   }
 
+  if $mysql::server::override_options and $mysql::server::override_options['mysqld'] and $mysql::server::override_options['mysqld']['socket'] {
+    $mysqlsocket = $mysql::server::override_options['mysqld']['socket']
+  } else {
+    $mysqlsocket = $options['mysqld']['socket']
+  }
+
+  exec { 'wait_for_mysql_socket_to_open':
+    command     => "ls ${mysqlsocket}",
+    tries       => '3',
+    try_sleep   => '10',
+    subscribe   => Service['mysqld'],
+    refreshonly => true,
```

Comment:
ok, your solution is more elegant. I tested it and the only things I changed are: 'test' is located in /usr/bin on ubuntu and centos. As I don't know if there is a distribution which places 'test' in /bin, I added path => '/bin:/usr/bin'. Also I changed subscribe to require, because without refreshonly, I see no reason to use subscribe.

Also how should I proceed? Should I close this pull request and make a completely new pull request with one squashed commit?